silk/tools/otns_manager.py

```diff
@@ -531,3 +642,80 @@ def unsubscribe_from_all_nodes(self):
     for subscriber in self.otns_monitor_map.values():
       subscriber.unsubscribe()
     self.otns_monitor_map.clear()
+
+  def update_layout(self):
+    """Update layout of nodes in auto layout mode.
+    """
+    if not self.auto_layout or not self.otns_node_map:
+      return
+
+    self.logger.debug("Updating nodes layout")
+    first_node = next(iter(self.otns_node_map))
+    center_x, center_y = first_node.device.get_otns_vis_layout_center()
+    radius = first_node.device.get_otns_vis_layout_radius()
+
+    leader_router = set()
+    children = set()
+    disabled_detached = set()
+    extaddr_to_node_map = {}
+
+    for node in self.otns_node_map.values():
+      extaddr_to_node_map[node.extaddr] = node
+      if node.role == RoleType.DISABLED or node.role == RoleType.DETACHED:
+        disabled_detached.add(node)
+      elif node.role == RoleType.CHILD:
+        children.add(node)
+      elif node.role == RoleType.ROUTER or node.role == RoleType.LEADER:
+        leader_router.add(node)
+
+    # filter out empty groups
+    groups = [leader_router, children, disabled_detached]
+    groups = [g for g in groups if g]
+
+    if not groups:
+      return
+
+    # order children near their parents
+    ordered_groups = []
+    for group in groups:
```

Comment:
Process each group of leader_router, children, disabled_detached separately, no need to put them into groups ?